src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -262,7 +262,13 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
                 var foundAwait = result.Any(pending => pending.Branch?.Kind == BoundKind.AwaitExpression);
                 if (!foundAwait)
                 {
-                    Diagnostics.Add(ErrorCode.WRN_AsyncLacksAwaits, currentMethodOrLambda.Locations[0]);
+                    // If we're on a LambdaSymbol, then use its 'DiagnosticLocation'.  That will be
+                    // much better than using its 'Location' (which is the entire span of the lambda).
+                    var diagnosticLocation = currentMethodOrLambda is LambdaSymbol lambda
```

Comment:
I'm on the fence about that.  The kind check worries me in that it tightly binds this specific kind to the type.  Normally is/as checks can be a concern.  But as this is only for the error path, i think it's totally ok to just have just the type-cast and remove all worry about the MethodKind/Type being in sync.